NOCList.playground/Contents.swift

```diff
@@ -29,56 +29,129 @@ This message will self destruct in 5 seconds.
+let maxRotGrab = (coverName: "Max RotGrab", realName: "Vanessa Redgrave", accessLevel: 4, compromised: false)
+let hannahWilliams = (coverName: "Hannah Williams", realName: "Ingeborga Dapkūnaitė", accessLevel: 5, compromised: true)
+let jackHarmon = (coverName: "Jack Harmon", realName: "Emilio Estevez", accessLevel: 6, compromised: true)
+let frankBarnes = (coverName: "Frank Barnes", realName: "Dale Dye", accessLevel: 9, compromised: false)
 
 
 
 //: ## Step 2
 //: Place the above constants inside an array. Declare this array as a constant as well.
-
+let easternEuropeTeam = [ethanHunt, jimPhelps, clairePhelps, eugeneKittridge, franzKrieger, lutherStickell, sarahDavies, maxRotGrab, hannahWilliams, jackHarmon, frankBarnes]
 
 
 //: ## Step 3
 //: Create a function that calculates the total number of compromised agents. Inside the function, iterate over the array of agents to determine which ones are compromised. Return the total count.
 
+func determineTotalCount() -> Int {
+    var totalCount = 0
+    for compromisedAgent in easternEuropeTeam {
+        if compromisedAgent.compromised == true {
+            totalCount += 1
+        }
+        
+    }
+    return totalCount
+}
 
 
 //: ## Step 4
 //: Call the above function to find the total number of compromised agents and then print a sentence that says "# agents have been compromised!" using string interpolation.
+print("\(determineTotalCount()) agents have been compromised!")
 
 
 
 //: ## Step 5
 //: Create a function called "findCleanAgents" that both prints the cover names of all uncompromised agents, as well as returns an array of agents that are uncompromised.
+func findCleanAgents() -> [Any] {
+    var cleanAgents = [Any]()
+    for uncompromisedAgents in easternEuropeTeam {
+        if uncompromisedAgents.compromised == false {
+        print(uncompromisedAgents.coverName)
+            cleanAgents.append(uncompromisedAgents)
+        }
+    }
+    return cleanAgents
+}
 
 
 
 //: ## Step 6
 //: Call the above function to find the total number of clean agents and print a message that says "# clean agents out of # total agents." Use the total number of agents in the array from step 2 as the second number in the string.
-
+let cleanAgentsArray = findCleanAgents()
+print("\(cleanAgentsArray.count) clean agents out of \(easternEuropeTeam.count)")
 
 
 //: ## Step 7
 //: Create a function called "findHighRisk" that prints out the real names and access levels of agents with level 8 or higher. If one of these agents is also currently compromised, add `**WARNING** **COMPROMISED**` to the end of the string that includes their name and access level.
 //: - Example: `Jon Voight, level: 9 **WARNING** **COMPROMISED**`
+func findHighRisk() {
+    var highRiskAgents = [Any]()
+    for agentsAtHighRisk in easternEuropeTeam {
+        if agentsAtHighRisk.accessLevel > 8 && agentsAtHighRisk.compromised == true {
```

Comment:
Your if-else statement looks great. Another way this could be worked out is through nil coalescing (something that will be covered in class later).

https://docs.swift.org/swift-book/LanguageGuide/BasicOperators.html